{
  "Comment": "ML Pipeline Orchestration",
  "StartAt": "DataIngestion",
  "States": {
    "DataIngestion": {
      "Type": "Task",
      "Resource": "arn:aws:lambda:us-east-1:361769570735:function:data_ingestion_lambda",
      "Next": "Preprocessing",
      "Retry": [
        {
          "ErrorEquals": [
            "States.ALL"
          ],
          "IntervalSeconds": 2,
          "MaxAttempts": 3,
          "BackoffRate": 2
        }
      ],
      "Catch": [
        {
          "ErrorEquals": [
            "States.ALL"
          ],
          "Next": "Failure"
        }
      ]
    },
    "Preprocessing": {
      "Type": "Task",
      "Resource": "arn:aws:lambda:us-east-1:361769570735:function:preprocessing_lambda",
      "Next": "AnalyzeSentiment",
      "Retry": [
        {
          "ErrorEquals": [
            "States.ALL"
          ],
          "IntervalSeconds": 2,
          "MaxAttempts": 3,
          "BackoffRate": 2
        }
      ],
      "Catch": [
        {
          "ErrorEquals": [
            "States.ALL"
          ],
          "Next": "Failure"
        }
      ]
    },
    "AnalyzeSentiment": {
      "Type": "Task",
      "Resource": "arn:aws:lambda:us-east-1:361769570735:function:sentiment_analysis",
      "Next": "Success",
      "Retry": [
        {
          "ErrorEquals": [
            "States.ALL"
          ],
          "IntervalSeconds": 2,
          "MaxAttempts": 3,
          "BackoffRate": 2
        }
      ],
      "Catch": [
        {
          "ErrorEquals": [
            "States.ALL"
          ],
          "Next": "Failure"
        }
      ]
    },
    "Success": {
      "Type": "Succeed"
    },
    "Failure": {
      "Type": "Fail",
      "Error": "WorkflowFailed",
      "Cause": "Step failed after retrying"
    }
  }
}